packages/language-php/grammars/tree-sitter/queries/highlights.scm

```diff
@@ -300,9 +332,9 @@
       "$" @punctuation.definition.variable.php
     ) @variable.parameter.php))
 
-; The "$bar" in `$foo->$bar`.
+; The "bar" and "$bar" in `$foo->bar` and `$foo->$bar`.
```

Comment:
Are `$foo->bar` and `$foo->$bar` interchangeable, or do they have different meanings? Are there any scenarios in which the `bar` in `$foo->bar` will be something other than a variable?

Reply:
After investigation, seems like the former is how you set instance variables and the latter is how you _dynamically_ set instance variables?

Do I have that right, and is that also the case for `Foo::bar` versus `Foo::$bar`?

Reply:
> `$foo->bar` and `$foo->$bar` ... the former is how you set instance variables and the latter is how you dynamically set instance variables?

Correct, except for the nitpick of "dynamic *access*", which could be getting as well as setting. Here's an example:
```php
class Foo {
    public $bar = 'a';
}

$foo = new Foo;
$foo->bar; // 'a'

$thud = 'bar';
$foo->$thud; // 'a'
```

> is that also the case for Foo::bar versus Foo::$bar?

No, static refs are a whole different beast. `Foo::bar` is a class constant reference, which I don't think has a corollary in object instances, while `Foo::$bar` is static variable ref, which is more comparable to `$foo->bar`. Another example:
```php
class Foo {
    const A = 1;
    static $a = 2;
    $b = 3;
}

Foo::A; // 1
Foo::$a; // 2
Foo::$b; // Access to undeclared static property Foo::$b.

$foo = new Foo;
$foo->a; // null ($a can only be accessed as a static var like Foo::$a)
$foo->b; // b
```

TMI?

Reply:
OK, that makes sense. I think that the way I've scoped them is still fine, but let me know if you think anything is weird on the main PR.

I did decide to scope the `bar` of `$foo->bar` as `support.other.property` rather than `variable.other.property` because I'm reluctant to scope anything as `variable` in PHP if it doesn't start with `$`.